py/htm/algorithms/anomaly_likelihood.py

```diff
@@ -15,662 +16,91 @@
     or five 9's, a log value is more useful for visualization, thresholding,
     etc.
     """
+    likelihood = self.get_likelihood(anomaly)
     # The log formula is:
     #     Math.log(1.0000000001 - likelihood) / Math.log(1.0 - 0.9999999999)
     return math.log(1.0000000001 - likelihood) / -23.02585084720009
-
-
-  @staticmethod
-  def _calcSkipRecords(numIngested, windowSize, learningPeriod):
-    """Return the value of skipRecords for passing to estimateAnomalyLikelihoods
-
-    If `windowSize` is very large (bigger than the amount of data) then this
-    could just return `learningPeriod`. But when some values have fallen out of
-    the historical sliding window of anomaly records, then we have to take those
-    into account as well so we return the `learningPeriod` minus the number
-    shifted out.
-
-    :param numIngested - (int) number of data points that have been added to the
-      sliding window of historical data points.
-    :param windowSize - (int) size of sliding window of historical data points.
-    :param learningPeriod - (int) the number of iterations required for the
-      algorithm to learn the basic patterns in the dataset and for the anomaly
-      score to 'settle down'.
-    """
-    numShiftedOut = max(0, numIngested - windowSize)
-    return min(numIngested, max(0, learningPeriod - numShiftedOut))
-
-
-  def anomalyProbability(self, value, anomalyScore, timestamp=None):
-    """
-    Compute the probability that the current value plus anomaly score represents
-    an anomaly given the historical distribution of anomaly scores. The closer
-    the number is to 1, the higher the chance it is an anomaly.
-
-    :param value: the current metric ("raw") input value, eg. "orange", or
-                   '21.2' (deg. Celsius), ...
-    :param anomalyScore: the current anomaly score
-    :param timestamp: [optional] timestamp of the ocurrence,
-                       default (None) results in using iteration step.
-    :returns: the anomalyLikelihood for this record.
-    """
-    if timestamp is None:
-      timestamp = self._iteration
-
-    dataPoint = (timestamp, value, anomalyScore)
-    # We ignore the first probationaryPeriod data points
-    if self._iteration < self._probationaryPeriod:
-      likelihood = 0.5
-    else:
-      # On a rolling basis we re-estimate the distribution
-      if ( (self._distribution is None) or
-           (self._iteration % self._reestimationPeriod == 0) ):
-
-        numSkipRecords = self._calcSkipRecords(
-          numIngested=self._iteration,
-          windowSize=self._historicalScores.maxlen,
-          learningPeriod=self._learningPeriod)
-
-        _, _, self._distribution = estimateAnomalyLikelihoods(
-          self._historicalScores,
-          skipRecords=numSkipRecords)
-
-      likelihoods, _, self._distribution = updateAnomalyLikelihoods(
-        [dataPoint],
-        self._distribution)
-
-      likelihood = 1.0 - likelihoods[0]
```

Comment:
One of these is the opposite of the other, despite having the same name.
:(

Reply:
yes, this naming was unfortunate